swarm/storage/filestore.go

```diff
@@ -83,16 +86,22 @@ func NewFileStore(store ChunkStore, params *FileStoreParams) *FileStore {
 // It returns a reader with the chunk data and whether the content was encrypted
 func (f *FileStore) Retrieve(ctx context.Context, addr Address) (reader *LazyChunkReader, isEncrypted bool) {
 	isEncrypted = len(addr) > f.hashFunc().Size()
-	getter := NewHasherStore(f.ChunkStore, f.hashFunc, isEncrypted)
+	getter := NewHasherStore(f.ChunkStore, f.hashFunc, isEncrypted, f.tags)
 	reader = TreeJoin(ctx, addr, getter, 0)
 	return
 }
 
 // Store is a public API. Main entry point for document storage directly. Used by the
 // FS-aware API and httpaccess
 func (f *FileStore) Store(ctx context.Context, data io.Reader, size int64, toEncrypt bool) (addr Address, wait func(context.Context) error, err error) {
-	putter := NewHasherStore(f.ChunkStore, f.hashFunc, toEncrypt)
-	return PyramidSplit(ctx, data, putter, putter)
+	putter := NewHasherStore(f.ChunkStore, f.hashFunc, toEncrypt, f.tags)
```

Comment:
a new hasherstore is even created for each file, why would you want to pass the tags object instead of just one tag?